### PR TITLE
1.19 Pull Req

### PR DESCRIPTION
**Pull request by Phelms215**
**Known Issues**
text translates do not work properly and display translate keys instead of proper language file info.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -116,3 +116,6 @@ run/
 
 # Avoid ignoring Gradle wrapper jar file (.jar files are usually ignored)
 !gradle-wrapper.jar
+
+gradle/
```
Comment (triphora) [inline .gitignore]: Don't remove the gradle folder.

File: README.md [hunk 2]
```diff
@@ -47,5 +47,4 @@ for the changes to take effect.
 [Discord Badge]: https://discord.com/assets/cb48d2a8d4991281d7a6a95d2f58195e.svg
 
 <!-- Hyperlink URLs -->
-[Fabric API Download]: https://modrinth.com/mod/fabric-api
-[Discord Invite]: https://discord.gg/D7TjNrArMw
+[Fabric API Download]: https://modrinth.com/mod/fabric-api
```
Comment (triphora) [inline README.md]: Why did you remove the Discord invite?

File: build.gradle [hunk 3]
```diff
@@ -1,77 +1,77 @@
...
 }
 
-sourceCompatibility = JavaVersion.VERSION_1_8
-targetCompatibility = JavaVersion.VERSION_1_8
+sourceCompatibility = JavaVersion.VERSION_17
+targetCompatibility = JavaVersion.VERSION_17
 
 archivesBaseName = project.archives_base_name
-version = "$project.mod_version-$project.minecraft_version"
+version = project.mod_version
```
Comment (triphora) [inline build.gradle:10]: Revert this

File: build.gradle [hunk 4]
```diff
@@ -1,77 +1,77 @@
...
+    // if it is present.
+    // If you remove this line, sources will not be generated.
+    withSourcesJar()
 }
 
 jar {
-    from "LICENSE"
+    from("LICENSE") {
+        rename { "${it}_${project.archivesBaseName}"}
+    }
```
Comment (triphora) [inline build.gradle:54]: Unnecessary

File: src/main/java/us/spaceclouds42/disableportals/DisablePortals.java [hunk 5]
```diff
@@ -1,21 +1,31 @@
...
+    public static final Logger LOGGER = LoggerFactory.getLogger("DisablePortals");
+
+
+
     @Override
     public void onInitialize() {
         CONF = Config.loadConfig(new File(FabricLoader.getInstance().getConfigDir() + "/DisablePortals.json"));
 
-        CommandRegistrationCallback.EVENT.register((dispatcher, dedicated) -> {
+        LOGGER.info("Initializing Mod");
```
Comment (triphora) [inline src/main/java/us/spaceclouds42/disableportals/DisablePortals.java]: Bunch of unnecessary stuff in this file, clean up the imports and the unnecessary newlines

File: src/main/java/us/spaceclouds42/disableportals/commands/DisablePortalsCommand.java [hunk 6]
```diff
@@ -123,10 +104,7 @@ private static int reloadConfig(ServerCommandSource source) {
         DisablePortals.CONF = Config.loadConfig(new File(FabricLoader.getInstance().getConfigDir() + "/DisablePortals.json"));
 
         source.sendFeedback(
-                new LiteralText("Reloaded the config file!").formatted(
-                        Formatting.GREEN
-                ),
-                false
+                Text.translatable("disableportals.reloadconfig").formatted(Formatting.GREEN), false
```
Comment (triphora) [inline src/main/java/us/spaceclouds42/disableportals/commands/DisablePortalsCommand.java]: Why did you convert these from literal texts to translatable texts? This is completely and entirely useless since translations are client-side

File: src/main/java/us/spaceclouds42/disableportals/mixin/EndGatewayBlockEntityMixin.java [hunk 7]
```diff
@@ -17,7 +20,7 @@ abstract class EndGatewayBlockEntityMixin {
             ),
             cancellable = true
     )
-    private void disableEndGateway(Entity entity, CallbackInfo ci) {
+    private static void disableEndGateway(World world, BlockPos pos, BlockState state, Entity entity, EndGatewayBlockEntity blockEntity, CallbackInfo ci) {
```
Comment (triphora) [inline src/main/java/us/spaceclouds42/disableportals/mixin/EndGatewayBlockEntityMixin.java:23]: ...does that work? That doesn't look like it would work.

File: src/main/java/us/spaceclouds42/disableportals/mixin/EndPortalBlockMixin.java [hunk 8]
```diff
@@ -22,6 +25,7 @@ abstract class EndPortalBlockMixin {
     )
     private void disableEndPortal(BlockState state, World world, BlockPos pos, Entity entity, CallbackInfo ci) {
         if (DisablePortals.CONF.main.disableEndPortals) {
+
```
Comment (triphora) [inline src/main/java/us/spaceclouds42/disableportals/mixin/EndPortalBlockMixin.java:25]: All of the changes in this file are unnecessary

File: src/main/java/us/spaceclouds42/disableportals/mixin/EnterBlockCriterionMixin.java [hunk 9]
```diff
@@ -22,6 +24,7 @@ abstract class EnterBlockCriterionMixin {
     )
     private void preventGatewayAdvancement(ServerPlayerEntity player, BlockState state, CallbackInfo ci) {
         if (state.getBlock() == Blocks.END_GATEWAY && DisablePortals.CONF.main.disableEndGateways) {
+
```
Comment (triphora) [inline src/main/java/us/spaceclouds42/disableportals/mixin/EnterBlockCriterionMixin.java:24]: All of the changes in this file are unnecessary

File: src/main/java/us/spaceclouds42/disableportals/mixin/NetherPortalBlockMixin.java [hunk 10]
```diff
@@ -3,6 +3,9 @@
 import net.minecraft.block.BlockState;
 import net.minecraft.block.NetherPortalBlock;
 import net.minecraft.entity.Entity;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
```
Comment (triphora) [inline src/main/java/us/spaceclouds42/disableportals/mixin/NetherPortalBlockMixin.java]: All of the changes in this file are unnecessary

File: src/main/resources/assets/disableportals/lang/en_us.json [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+  "disableportals.end_enable" : "Enabled End portals",
+  "disableportals.end_disable" : "Disabled End portals",
+  "disableportals.end_disabled_notice" : "The End is disabled, you are not able to enter!",
+
+  "disableportals.endgw_enable" : "Enabled End Gateways",
+  "disableportals.endgw_disable" : "Disabled End Gateways",
+  "disableportals.endgw_disabled_notice" : "The End Gateways are disabled, you are not able to enter!",
+
+  "disableportals.reloadconfig" : "Reloaded DisablePortals Configuration File!"
+}
```
Comment (triphora) [inline src/main/resources/assets/disableportals/lang/en_us.json]: Can't be used on the server

File: src/main/resources/fabric.mod.json [hunk 12]
```diff
@@ -5,7 +5,8 @@
   "name": "Disable Portals",
   "description": "Adds a simple command for disabling nether and end portals separately.",
   "authors": [
-    "SpaceClouds42"
+    "SpaceClouds42",
+    "PHELMS"
```
Comment (triphora) [inline src/main/resources/fabric.mod.json:11]: You're not a primary author of this mod, you should go into the `contributors` array

File: src/main/resources/fabric.mod.json [hunk 13]
```diff
@@ -22,7 +23,9 @@
     "disableportals.mixins.json"
   ],
   "depends": {
-    "fabricloader": ">=0.11.2",
-    "minecraft": "1.16.5"
+    "fabricloader": ">=0.14.6",
+    "fabric": "*",
+    "minecraft": "~1.19",
+    "java": ">=17"
```
Comment (triphora) [inline src/main/resources/fabric.mod.json:31]: Why did you add a dependency on Fabric API?

**Review by triphora — CHANGES_REQUESTED** [review]
I'm sorry to say, but this PR is a complete mess. Here are some pointers to hopefully get it in a better shape.

File: src/main/resources/fabric.mod.json (hunk 13 above)
Comment (Phelms215) [reply src/main/resources/fabric.mod.json:31]: The mod itself declared a dependency on the api - I made it official

File: src/main/java/us/spaceclouds42/disableportals/mixin/EndGatewayBlockEntityMixin.java (hunk 7 above)
Comment (Phelms215) [reply src/main/java/us/spaceclouds42/disableportals/mixin/EndGatewayBlockEntityMixin.java:23]: tryTeleportingEntity is now static

File: src/main/java/us/spaceclouds42/disableportals/commands/DisablePortalsCommand.java (hunk 6 above)
Comment (Phelms215) [reply src/main/java/us/spaceclouds42/disableportals/commands/DisablePortalsCommand.java]: Didn't realize there was no way (without this fabric language hack mod) to get translations server side..Kind of lame admins can't get a command in their native language sent to them.. guess I'll revert to the previous literals

**Comment by Phelms215** [comment]
Closing to resubmit from upstream branch
